### PR TITLE
BL-12193 Decode access token

### PR DESCRIPTION
**Pull request by JoshuaLicense**
| Q                | A
| ---------------- | ---
| Type             | Feature
| Ticket           | BL-12193

**Description of the change:**

This MR:
- Will add ability to decode JWT, thereby validating the token is valid.
- Add other unimplemented methods to ensure the register flow works.

### CODE REVIEW

**Comment by JoshuaLicense** [comment]
Please hold off merging this one before we get a green action build - This should go green once the `dvsa/contracts` package is merged.

File: src/Client.php
Comment (Hodgy) [inline src/Client.php:115]: Shouldn't this be ` 'Permanent' => $permanent`

**Review by Hodgy — CHANGES_REQUESTED** [review]

**Pull request merged**
